src/contracts/interfaces/IPaymentCoordinator.sol

```diff
@@ -0,0 +1,175 @@
+        uint32 rootIndex;
+        // proof of the claimaint's account root in the Merkle tree
+        uint32 accountIndex;
+        bytes accountTreeProof;
+        // The indices and proofs of the leafs in the claimaint's merkle tree for this root
+        uint32[] leafIndices;
+        bytes[] tokenTreeProofs;
+    }
+
+    /// EVENTS ///
+
+    event RangePaymentCreated(address indexed avs, bytes32 rangePaymentHash, RangePayment rangePayment);
+    event RangePaymentForAllCreated(address indexed submitter, bytes32 rangePaymentHash, RangePayment rangePayment);
+    event PaymentUpdaterSet(address indexed oldPaymentUpdater, address indexed newPaymentUpdater);
+    event PayAllForRangeSubmitterSet(address indexed payAllForRangeSubmitter, bool indexed oldValue, bool indexed newValue);
+    event ActivationDelaySet(uint64 oldActivationDelay, uint64 newActivationDelay);
+    event CalculationIntervalSecondsSet(uint64 oldCalculationIntervalSeconds, uint64 newCalculationIntervalSeconds);
+    event GlobalCommissionBipsSet(uint16 oldGlobalCommissionBips, uint16 newGlobalCommissionBips);
+    event RecipientSet(address indexed account, address indexed recipient);
+    event RootSubmitted(
+        bytes32 root,
+        uint32 paymentCalculationStartTimestamp,
+        uint32 paymentCalculationEndTimestamp,
+        uint32 activatedAt
+    );
+    event PaymentClaimed(ClaimsTreeMerkleLeaf leaf);
+
+    /// VIEW FUNCTIONS ///
+
+    /// @notice The address of the entity that can update the contract with new merkle roots
+    function paymentUpdater() external view returns (address);
+
+    /**
+     * @notice The interval in seconds at which the calculation for range payment distribution is done.
+     * @dev Payment durations must be multiples of this interval.
+     */
+    function calculationIntervalSeconds() external view returns (uint64);
+
+    /// @notice The maximum amount of time that a range payment can end in the future
+    function MAX_PAYMENT_DURATION() external view returns (uint64);
+
+    /// @notice The lower bound for the start of a range payment
+    function LOWER_BOUND_START_RANGE() external view returns (uint64);
+
+    /// @notice Delay in timestamp before a posted root can be claimed against
+    function activationDelay() external view returns (uint64);
+
+    /// @notice Mapping: account => the address of the entity to which new payments are directed on behalf of the account
+    function recipientOf(address account) external view returns (address);
+
+    /// @notice Mapping: recipient => token => total amount claimed
+    function cumulativeClaimed(address recipient, IERC20 token) external view returns (uint256);
+
+    /// @notice the commission for all operators across all avss
+    function globalOperatorCommissionBips() external view returns (uint16);
+
+    /// @notice returns the hash of the leaf
+    function calculateLeafHash(ClaimsTreeMerkleLeaf calldata leaf) external view returns (bytes32);
```

Comment:
One thing that helps here is *why* does this method exist? Who would use it?

I think its because clients use it to prepare the request for claiming their rewards. Right?

Reply:
Yes and allow forward-compatability if this is being used as part of offchain merkleization. Will add a comment for that.